pkg/runtime/reconciler.go

```diff
@@ -604,27 +636,25 @@ func (r *resourceReconciler) patchResourceMetadataAndSpec(
 	}
 
 	rlog.Enter("kc.Patch (metadata + spec)")
-	// Save a copy of the latest object, to reset 'Status' after performing
-	// the kc.Patch() operation
-	latestCopy := latest.DeepCopy()
-	err = r.kc.Patch(
-		ctx,
-		latest.RuntimeObject(),
-		client.MergeFrom(desired.DeepCopy().RuntimeObject()),
-	)
-	// Reset the status of latest object after patching.
-	latest.SetStatus(latestCopy)
-	rlog.Exit("kc.Patch (metadata + spec)", err)
-
-	if err != nil {
-		return err
+	dobj := desired.DeepCopy().RuntimeObject()
+	lobj := latest.DeepCopy().RuntimeObject()
+	patch := client.MergeFrom(dobj)
+	err = r.kc.Patch(ctx, lobj, patch)
+	if err == nil {
+		if rlog.IsDebugEnabled() {
```

Comment:
Why is this condition necessary? Can we not just print to Debug and let the logger determine the log level of the output?

Reply:
It's necessary because `getPatchDocument()` is a relatively expensive operation (marshals, unmarshals, marshals again JSON) and there's no point executing it if debug logging isn't enabled.